File: Lib/Telerik/Examples/QuickStart/RangeSelector/MSFTStocks/StringLimiterFormatProvider.vb
Imports System.Linq
Imports System.Text

Namespace Telerik.Examples.WinControls.RangeSelector.MSFTStocks
	Friend Class StringLimiterFormatProvider
		Implements IFormatProvider, ICustomFormatter
		Public Function GetFormat(ByVal formatType As Type) As Object Implements IFormatProvider.GetFormat
			Return Me
		End Function

'INSTANT VB NOTE: The parameter format was renamed since Visual Basic will not allow parameters with the same name as their enclosing function or property:
		Public Function Format(ByVal format_Renamed As String, ByVal arg As Object, ByVal formatProvider As IFormatProvider) As String Implements ICustomFormatter.Format
			Dim s As String = arg.ToString()
			If s IsNot Nothing Then
				If s.Length = 1 Then
					Return "0"
				End If

				Return s.Remove(s.Length - 6)
			Else
				Return Nothing
			End If
		End Function
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/RangeSelector/MSFTStocks/StringLimiterFormatProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Telerik.Examples.WinControls.RangeSelector.MSFTStocks
{
    class StringLimiterFormatProvider : IFormatProvider, ICustomFormatter
    {
        public object GetFormat(Type formatType)
        {
            return this;
        }

        public string Format(string format, object arg, IFormatProvider formatProvider)
        {
            string s = arg.ToString();
            if (s != null)
            {
                if (s.Length == 1)
                {
                    return "0";
                }

                return s.Remove(s.Length - 6);
            }
            else 
            {
                return null;
            }
        }
    }
}
